controller/eventController.js

```diff
@@ -146,13 +149,40 @@ const update = (req, res) => {
     if (!user || !ValidationUtil.isUserAdmin(user.accessType)) {
         return res.status(HttpStatus.UNAUTHORIZED).json({
             message: 'Unauhorized.'
-        })
+        });
     }
 
     Event.findOneAndUpdate({ _id: new ObjectId(eventId), companyID: new ObjectId(user.companyID) },
         req.body, { runValidators: true, new: true }).then(event => {
             if (event) {
-                return res.status(HttpStatus.OK).json(event);
+                // find event participants
+                Invite.find({ eventID: event._id, responseType: ResponseTypeEnum['Accepted'] }, { userID: 1 }).then(invites => {
+                    const userIDs = invites.map(invite => invite.userID);
+                    // get user mails
+                    User.find({ _id: { $in: userIDs } }, { email: 1 }).then(users => {
+                        const emails = users.map(u => u.email);
+                        // now get admin mails for cc - this is done separately since not all admins are part of all events and 
+                        // the idea at the moment is to include all admins in the mail as cc
+                        User.find({ companyID: user.companyID, accessType: { $in: [UserTypeEnum['Admin'], UserTypeEnum['Super-Admin']] } }, { email: 1 }).then(adminUsers => {
+                            const adminEmails = adminUsers.map(adminUser => adminUser.email);
+                            if (emailUtil.sendEmail(`${event.name} - Update`, 'Event details have been updated. Check out what\' changed.', emails, adminEmails)) {
+                                return res.status(HttpStatus.OK).json(event);
+                            } else {
+                                // what to do? update went fine, but fails at sending emails
+                                return res.status(HttpStatus.OK).json(event);
```

Comment:
But the update went fine. It is the sending of the notification that the update went through that fails. It makes little sense to send an `INTERNAL_SERVER_ERROR` and to say it failed, even though the update went fine.

I am thinking maybe we could return a `2XX` status code, but the message could say `Could not send event update notification through mail.`. @djuricic1 is there a viable status code to be used here?

Reply:
@SavoX2 by `INTERNAL_SERVER_ERROR` I was referring to the mail server. If a more suitable status code exist then you can add it, but regardless the status code, you should provide a message in response

Reply:
At the moment no `message` attribute exists in any response that is completed successfully. 

I have an idea, but it would require a bit of reformating of existing code. The idea, that we shunned off 2 weeks back, is to create a response object class as follows: 
`Response::{`
  `value:` the json returned if update/get/create/whatever was successful, 
  `message:` this field will be filled in case of error, or in the obscure case that as this, where the primary succeeds, but an additional action fails like in this case, status code 202 or 206 perhaps
`}`. 
This `Response` object would then be created and returned in every function.

Reply:
I have changed the return values here, but a separated task should be created to refactor the entire app code return values, not just the ones here.